Controller/Adminhtml/Account/Delete.php

```diff
@@ -38,100 +38,78 @@
-    private NostoIframeMetaBuilder $nostoIframeMetaBuilder;
     private NostoHelperScope $nostoHelperScope;
     private NostoHelperCache $nostoHelperCache;
 
     /**
      * @param Context $context
      * @param NostoHelperAccount $nostoHelperAccount
-     * @param NostoIframeMetaBuilder $nostoIframeMetaBuilder
      * @param NostoCurrentUserBuilder $nostoCurrentUserBuilder
      * @param NostoHelperScope $nostoHelperScope
      * @param NostoHelperCache $nostoHelperCache
-     * @param Json $result
      */
     public function __construct(
         Context $context,
         NostoHelperAccount $nostoHelperAccount,
-        NostoIframeMetaBuilder $nostoIframeMetaBuilder,
         NostoCurrentUserBuilder $nostoCurrentUserBuilder,
         NostoHelperScope $nostoHelperScope,
-        NostoHelperCache $nostoHelperCache,
-        Json $result
+        NostoHelperCache $nostoHelperCache
     ) {
         parent::__construct($context);
 
-        $this->nostoIframeMetaBuilder = $nostoIframeMetaBuilder;
         $this->nostoHelperAccount = $nostoHelperAccount;
-        $this->result = $result;
         $this->nostoCurrentUserBuilder = $nostoCurrentUserBuilder;
         $this->nostoHelperScope = $nostoHelperScope;
         $this->nostoHelperCache = $nostoHelperCache;
     }
 
     /**
-     * @return Json
+     * @return Redirect
+     * @suppress PhanUndeclaredMethod
+     * @noinspection PhpPossiblePolymorphicInvocationInspection
      * @throws Exception
      */
     public function execute()
     {
+        $resultRedirect = $this->resultFactory->create(ResultFactory::TYPE_REDIRECT);
         $storeId = $this->_request->getParam('store');
         $store = $this->nostoHelperScope->getStore($storeId);
+
         if ($store === null) {
             throw new LocalizedException(new Phrase('No account found'));
         }
 
         $account = $this->nostoHelperAccount->findAccount($store);
+
         if ($account !== null) {
             $currentUser = $this->nostoCurrentUserBuilder->build();
             if ($this->nostoHelperAccount->deleteAccount($account, $store, $currentUser)) {
                 //Invalidate the cache
                 $this->nostoHelperCache->invalidatePageCache();
                 $this->nostoHelperCache->invalidateLayoutCache();
 
-                $response = [];
-                $response['success'] = true;
-                $response['redirect_url'] = IframeHelper::getUrl(
-                    $this->nostoIframeMetaBuilder->build($store),
-                    null, // we don't have an account anymore
-                    $this->nostoCurrentUserBuilder->build(),
-                    [
-                        'message_type' => Nosto::TYPE_SUCCESS,
-                        'message_code' => Nosto::CODE_ACCOUNT_DELETE,
-                    ]
+                $this->getMessageManager()->addSuccessMessage(
+                    "Nosto has been successfully disconnected from the store."
```

Comment:
We don't have multiple languages in our M2 module, but I could modify the code to use translate function, even though it will require adding a few inspection suppressions since `addSuccessMessage` function accepts string as a parameter.

Reply:
@olsi-qose IIRC the wrong domain message is translated to at least French. Do you remember how was that done?